.cloud-build/cleanup/resource_cleanup_manager.py

```diff
@@ -156,3 +165,46 @@ class BatchPredictionJobCleanupManager(VertexAIResourceCleanupManager):
 
 class ExperimentCleanupManager(VertexAIResourceCleanupManager):
     vertex_ai_resource = aiplatform.Experiment
+
+class BucketCleanupManager(ResourceCleanupManager):
+    vertex_ai_resource = storage.bucket.Bucket
+
+    def list(self) -> Any:
+        storage_client = storage.Client()
+        #return [ bucket for bucket in storage_client.list_buckets()]
+        return list(storage_client.list_buckets())
+
+    def delete(self, resource):
+        try:
+            resource.delete(force=True)
+        except Exception as e:
+            print(e)
+
+    @property
+    def type_name(self) -> str:
+        return str(type(self.vertex_ai_resource))
+
+    def get_seconds_since_modification(self, resource: Any) -> float:
+        # Bucket has no last_update property, only time created
+        created_time = resource.time_created
+        current_time = DatetimeWithNanoseconds.now()
+        return float(current_time.timestamp() - created_time.timestamp())
+
+    def resource_name(self, resource: Any) -> str:
+        return resource.name
+
+    def is_deletable(self, resource: Any) -> bool:
+        time_difference = self.get_seconds_since_modification(resource)
+
+        if not 'your-bucket-name' in self.resource_name(resource):
+            print(f"Skipping '{resource}' not a Vertex AI notebook bucket")
```

Comment:
Maybe this sounds better? 

"Skipping '{resource}' as it is not a Vertex AI notebook bucket"